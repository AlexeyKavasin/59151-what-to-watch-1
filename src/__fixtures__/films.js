export default [
  {
    name: `Fantastic Beasts`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    id: 0,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Comedy`
  },
  {
    name: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    id: 1,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Comedy`
  },
  {
    name: `Macbeth`,
    poster: `img/macbeth.jpg`,
    id: 2,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Comedy`
  },
  {
    name: `Aviator`,
    poster: `img/Aviator.jpg`,
    id: 3,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Comedy`
  },
  {
    name: `Pulp Fiction`,
    poster: `img/pulp-fiction.jpg`,
    id: 4,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isPlaying: false,
    genre: `Documentary`
  }
];
