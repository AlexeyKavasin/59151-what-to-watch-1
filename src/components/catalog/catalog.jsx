import React from "react";
import PropTypes from "prop-types";
import GenresList from "../genres-list/genreslist.jsx";
import MovieList from "../movie-list/movielist.jsx";

class Catalog extends React.PureComponent {
  render() {
    return (
      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <GenresList/>
        <MovieList/>
        <div className="catalog__more">
          <button className="catalog__button" type="button">Show more</button>
        </div>
      </section>
    );
  }
}

Catalog.propTypes = {
  onGenreChange: PropTypes.func
};

export {Catalog};
export default Catalog;
