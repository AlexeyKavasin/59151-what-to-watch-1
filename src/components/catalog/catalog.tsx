import * as React from "react";
import {connect} from "react-redux";
import GenresList from "../genres-list/genreslist";
import MovieList from "../movie-list/movielist";
import {getCurrentGenre, getAllGenres, filterFilmsByGenre} from "../../redux/reducer/data/selectors.js";
import {withActiveCard} from "../../hocs/with-active-card/with-active-card";
import {selectGenre} from "../../redux/reducer/actions";
import {ICatalog} from "../../interfaces";

const MovieListWithActiveCard = withActiveCard(MovieList);

class Catalog extends React.PureComponent<ICatalog, null> {
  render() {
    const {films, genres, onGenreChange, currentGenre} = this.props;
    return (
      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <GenresList genres={genres} currentGenre={currentGenre} onGenreChange={onGenreChange}/>
        <MovieListWithActiveCard films={films}/>
        <div className="catalog__more">
          <button className="catalog__button" type="button">Show more</button>
        </div>
      </section>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  ownProps,
  currentGenre: getCurrentGenre(state),
  genres: getAllGenres(state),
  films: filterFilmsByGenre(state)
});

const mapDispatchToProps = (dispatch) => ({
  onGenreChange: (genre) => {
    dispatch(selectGenre(genre));
  }
});

export {Catalog};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Catalog);
