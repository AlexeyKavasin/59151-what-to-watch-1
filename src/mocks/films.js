export default [
  {
    name: `Fantastic Beasts`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    name: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`
  },
  {
    name: `Macbeth`,
    src: `img/macbeth.jpg`
  },
  {
    name: `Aviator`,
    src: `img/Aviator.jpg`
  },
  {
    name: `Pulp Fiction`,
    src: `img/pulp-fiction.jpg`
  }
];
