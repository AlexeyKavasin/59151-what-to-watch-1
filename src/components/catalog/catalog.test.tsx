import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Catalog} from './catalog';
import films from '../../__fixtures__/films';

jest.mock(`../genres-list/genreslist`, () => () => `GenresList`);
jest.mock(`../movie-list/movielist`, () => () => `MovieList`);

describe(`Catalog tests`, () => {
  it(`Catalog renders correctly`, () => {
    const genres = [`All genres`, `Comedy`, `Documentary`];
    const currentGenre = `All genres`;
    const onGenreChange = jest.fn;
    const tree = renderer
      .create(<Catalog onGenreChange={onGenreChange} films={films} currentGenre={currentGenre} genres={genres}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
