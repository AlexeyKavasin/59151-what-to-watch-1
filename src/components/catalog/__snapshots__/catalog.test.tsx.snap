// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog tests Catalog renders correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <GenresList
    currentGenre="All genres"
    genres={
      Array [
        "All genres",
        "Comedy",
        "Documentary",
      ]
    }
    onGenreChange={[Function]}
  />
  <MovieList
    activeCard={-1}
    films={
      Array [
        Object {
          "genre": "Comedy",
          "id": 0,
          "isPlaying": false,
          "name": "Fantastic Beasts",
          "poster": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
          "trailer": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        },
        Object {
          "genre": "Comedy",
          "id": 1,
          "isPlaying": false,
          "name": "Bohemian Rhapsody",
          "poster": "img/bohemian-rhapsody.jpg",
          "trailer": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        },
        Object {
          "genre": "Comedy",
          "id": 2,
          "isPlaying": false,
          "name": "Macbeth",
          "poster": "img/macbeth.jpg",
          "trailer": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        },
        Object {
          "genre": "Comedy",
          "id": 3,
          "isPlaying": false,
          "name": "Aviator",
          "poster": "img/Aviator.jpg",
          "trailer": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        },
        Object {
          "genre": "Documentary",
          "id": 4,
          "isPlaying": false,
          "name": "Pulp Fiction",
          "poster": "img/pulp-fiction.jpg",
          "trailer": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        },
      ]
    }
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;
